dicompylercore/dicomparser.py

```diff
@@ -625,6 +626,20 @@ def GetStructureCoordinates(self, roi_number):
                             plane['data'] = \
                                 self.GetContourPoints(c.ContourData)
 
+                            # Verify if all z-coordinates in this plane
+                            # are identical
+                            z0 = c.ContourData[2]
+                            if not all(
+                                z == z0
+                                for z in islice(c.ContourData, 5, None, 3)
+                            ):
+                                print(
+                                    f"Warning ROI Number {roi_number}:\n"
+                                    "Not all z-coordinates are identical.\n"
+                                    "Execution halted."
+                                )
+                                return {}
+
```

Comment:
Consider using `logger.debug` instead of `print` as that is used throughout the rest of dicompyler-core as well. That way upstream users can catch logs in one place.